pkg/compute/executor.go

```diff
@@ -392,6 +394,17 @@ func (e *BaseExecutor) Run(ctx context.Context, state store.LocalExecutionState)
 		}
 	}
 
+	// There can be a scenario where the execution may already have been cancelled
+	// by the user. In such case we should not mark it as completed.
+	currentExecutionState, err := e.store.GetExecution(ctx, execution.ID)
+	if err != nil {
+		return err
+	}
+	if currentExecutionState.State == store.ExecutionStateCancelled {
+		log.Ctx(ctx).Debug().Msgf("Execution has already been cancelled for %s", execution.ID)
+		return nil
+	}
+
```

Comment:
do we still need this check since cancelling the execution would trigger an `ErrAlreadyCancelled`